State machines with capitalized names fail when referenced within other state machines

Originally reported on Google Code with ID 695

---

**What steps will reproduce the problem?**
1. Try the following umple code with two state machines - sm and Other:

``` java
class A 
{
  sm 
  {
    s1 
    {
      entry / {Other = Other.os2;}
    }
  }

  Other
  {
    os1
    {

    }
    os2
    {

    }
  }
}
```

```
Notice that entry into s1 of sm causes other to move into state os2.
```
1. Generate Java code for this Umple model.
2. Compile the Java code.

The Java compilation fails, with the following error:

```
A.java:60: error: cannot find symbol
      Other = Other.os2;
      ^
  symbol:   variable Other
  location: class A
1 error
```

This code does not fail Java compilation if the second state machine is named with
a lowercase name (ie "other") nor does it fail if the first state machine ("sm") does
not reference the second state machine (ie. if the entry action is removed).

This bug is present in the "Traffic Lights B" example on UmpleOnline.

**Please use labels and text to provide additional information.**

---

Reported by @CraigBryan on 2015-03-31 13:55:59

--- a/umpleonline/ump/TrafficLightsB.ump
+++ b/umpleonline/ump/TrafficLightsB.ump
@@ -3,14 +3,14 @@ class IntersectionTimerDriven {
   
   // Traffic lights are green for 25s in one direction
   // and 20s in the other. 
-  Integer directionOneGoTime=25000;
-  Integer directionTwoGoTime=20000;
+  Integer directionOneGoTime = 25;
+  Integer directionTwoGoTime = 20;
 
   // Lights are yellow for 5s
-  Integer yellowTime=5000;
+  Integer yellowTime = 5;
 
   // Pedestrians have 10s extra warning
-  Integer extraPedestrianWarning=10000;  
+  Integer extraPedestrianWarning = 10;  
   
   flow {
     FourWayStop 

--- a/umpleonline/ump/TrafficLightsB.ump
+++ b/umpleonline/ump/TrafficLightsB.ump
@@ -16,38 +16,38 @@ class IntersectionTimerDriven {
     FourWayStop 
     {
       entry / {
-        directionOneLight = DirectionOneLight.FlashingRed;
-        directionTwoLight = directionTwoLight.FlashingRed;
+        setDirectionOneLight(DirectionOneLight.FlashingRed);
+        setDirectionTwoLight(DirectionTwoLight.FlashingRed);
       }
       initiateRegularOperation -> DirectionOneGo;    
     }
     DirectionOneGo
     {
       entry / {
-        directionOneLight = directionOneLight.Green;
-        directionTwoLight = directionTwoLight.Red;
+        setDirectionOneLight(DirectionOneLight.Green);
+        setDirectionTwoLight(DirectionTwoLight.Red);
       }
       after(directionOneGoTime) -> DirectionOneHalting;
       detectMalfunction -> FourWayStop;
     }
     DirectionOneHalting
     {
-      entry / {directionOneLight = directionOneLight.Yellow; }
+      entry / {setDirectionOneLight(DirectionOneLight.Yellow);}
       after(yellowTime) -> DirectionTwoGo;
       detectMalfunction -> FourWayStop;
     }
     DirectionTwoGo
     {
       entry / {
-        directionOneLight = directionOneLight.Red;
-        directionTwoLight = directionTwoLight.Green;
+        setDirectionOneLight(DirectionOneLight.Red);
+        setDirectionTwoLight(DirectionTwoLight.Green);
       }
       after(directionOneGoTime) -> DirectionTwoHalting;
       detectMalfunction -> FourWayStop;
     }
     DirectionTwoHalting 
     {
-      entry / {directionTwoLight = directionTwoLight.Yellow; }
+      entry / {setDirectionTwoLight(DirectionTwoLight.Yellow);}
       after(yellowTime) -> DirectionOneGo;
       detectMalfunction -> FourWayStop;
     }        
@@ -57,37 +57,40 @@ class IntersectionTimerDriven {
   {
     FourWayStop
     {
-      entry / {P1Light = P1Light.DoNotWalk;  P2Light = P2Light.DoNotWalk;}
+      entry / {
+	setP1Light(P1Light.DoNotWalk);  
+	setP2Light(P2Light.DoNotWalk);
+      }
       initiateRegularOperation -> P2Go;
     }
     P1Go
     {
-      entry / {P1Light = P1Light.Walk;}
+      entry / {setP1Light(P1Light.Walk);}
       after(directionOneGoTime-extraPedestrianWarning)-> P1Halting;
       detectMalfunction -> FourWayStop;
     }
     P1Halting
     {
-      entry / {P1Light = P1Light.Flashing;}
+      entry / {setP1Light(P1Light.Flashing);}
       after(extraPedestrianWarning+yellowTime) -> P2Go;
       detectMalfunction -> FourWayStop;
     }
     P2Go
     {
-      entry / {P2Light = P2Light.Walk;}
+      entry / {setP2Light(P2Light.Walk);}
       after(directionTwoGoTime-extraPedestrianWarning) -> P2Halting;
       detectMalfunction -> FourWayStop;
     }
     P2Halting
     {
-      entry / {P2Light = P2Light.Flashing;}
+      entry / {setP2Light(P2Light.Flashing);}
       after(extraPedestrianWarning+yellowTime)  -> P1Go;
       detectMalfunction -> FourWayStop;
     }
   }
   
   
-  directionOneLight 
+  DirectionOneLight 
   {
     FlashingRed {} 
     Red {}
@@ -95,7 +98,7 @@ class IntersectionTimerDriven {
     Green {}    
   }
   
-  directionTwoLight
+  DirectionTwoLight
   {
     FlashingRed {}
     Red {}
@@ -117,5 +120,3 @@ class IntersectionTimerDriven {
   }
 }
 //$?[End_of_model]$?
-// @@@skipcompile - will not compile until issue 695 is fixed
-// @@@skiprubycompile - incorrect ruby code generated until issue 694 fixed

